webui/react/src/hooks/useSettings.ts

```diff
@@ -271,27 +281,41 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
 
         newSettings[setting as keyof T] = defaultSetting.defaultValue;
       });
+      stateOb.update((s) => s.set(config.storagePath, newSettings));
 
-      update(config.storagePath, newSettings);
-
-      await updateDB(newSettings);
+      await updateDB(newSettings, user);
 
       navigate('', { replace: true });
     },
-    [config, update, updateDB, navigate, settings],
+    [config, updateDB, navigate, settings, user, stateOb],
   );
 
   const updateSettings = useCallback(
-    async (updates: Settings, shouldPush = false) => {
-      if (!settings) return;
-
-      const newSettings = { ...settings, ...updates };
-
-      if (isEqual(newSettings, settings)) return;
-
-      update(config.storagePath, newSettings);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (updates: Settings, shouldPush = false) => {
+      stateOb.update((s) =>
+        s.set(
+          config.storagePath,
+          s.get(config.storagePath) === updates
+            ? s.get(config.storagePath) ?? {}
+            : { ...s.get(config.storagePath), ...updates },
+        ),
+      );
+    },
+    [config, stateOb],
+  );
 
-      await updateDB(newSettings);
+  useEffect(() => {
+    return derivedOb.subscribe(async (cur, prev) => {
+      if (
+        !cur ||
+        !prev ||
+        !user ||
+        cur === prev ||
+        Object.keys(cur).length !== Object.keys(prev).length
+      )
+        return;
```

Comment:
These are to prevent unnecessary `POST` when loading, can we assume settings would always have the same set of keys when loaded